mate-panel/libpanel-util/panel-session-manager.c

```diff
@@ -42,136 +49,102 @@ panel_session_manager_class_init (PanelSessionManagerClass *klass)
 
 	object_class = G_OBJECT_CLASS (klass);
 
-	object_class->constructor = panel_session_manager_constructor;
+	object_class->finalize = panel_session_manager_finalize;
 }
 
 static void
 panel_session_manager_init (PanelSessionManager *manager)
 {
-}
-
-static GObject *
-panel_session_manager_constructor (GType                  type,
-				   guint                  n_construct_properties,
-				   GObjectConstructParam *construct_properties)
-{
-	GObject *obj;
-	GError  *error;
-
-	obj = G_OBJECT_CLASS (panel_session_manager_parent_class)->constructor (
-							type,
-							n_construct_properties,
-							construct_properties);
-
-
-	panel_dbus_service_define_service (PANEL_DBUS_SERVICE (obj),
-					   "org.gnome.SessionManager",
-					   "/org/gnome/SessionManager",
-					   "org.gnome.SessionManager");
-
-	error = NULL;
-	if (!panel_dbus_service_ensure_connection (PANEL_DBUS_SERVICE (obj),
-						   &error)) {
-		g_message ("Could not connect to session manager: %s",
-			   error->message);
+	GError *error = NULL;
+
+	manager->proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
+							G_DBUS_PROXY_FLAGS_NONE,
+							NULL,
+							"org.gnome.SessionManager",
+							"/org/gnome/SessionManager",
+							"org.gnome.SessionManager",
+							NULL,
+							&error);
+	if (manager->proxy == NULL)
+	{
+		g_warning ("Unable to contact session manager daemon: %s\n", error->message);
 		g_error_free (error);
 	}
-
-	return obj;
 }
 
 void
 panel_session_manager_request_logout (PanelSessionManager           *manager,
 				      PanelSessionManagerLogoutType  mode)
 {
-	GError *error;
-	DBusGProxy *proxy;
+	GError *error = NULL;
+	GVariant *_ret;
 
 	g_return_if_fail (PANEL_IS_SESSION_MANAGER (manager));
-
-	error = NULL;
-
-	if (!panel_dbus_service_ensure_connection (PANEL_DBUS_SERVICE (manager),
-						   &error)) {
-		g_warning ("Could not connect to session manager: %s",
-			   error->message);
-		g_error_free (error);
-		return;
-	}
-
-	proxy = panel_dbus_service_get_proxy (PANEL_DBUS_SERVICE (manager));
-
-	if (!dbus_g_proxy_call (proxy, "Logout", &error,
-				G_TYPE_UINT, mode, G_TYPE_INVALID,
-				G_TYPE_INVALID) &&
-	    error != NULL) {
+	g_return_if_fail (manager->proxy != NULL);
```

Comment:
`g_return_if_fail` is meant for programmer errors:
https://developer.gnome.org/glib/stable/glib-Warnings-and-Assertions.html#g-return-if-fail

Reply:
?